src/time_tracker/invoices/handlers.clj

```diff
@@ -94,3 +95,34 @@
   "Retrieves all invoices"
   [request connection]
   (res/response (invoices-db/retrieve-all connection)))
+
+;; Endpoint for retrieving a single invoice
+;; GET /api/invoices/<id>/
+(defn retrieve
+  [request connection]
+  (let [invoice-id (Integer/parseInt (:id (:route-params request)))]
+    (if-let [invoice (invoices-db/retrieve
+                      connection
+                      invoice-id)]
+      (do
+        (log/debug {:event "invoice-received" :data invoice})
+        (res/response invoice))
+      web-util/error-not-found)))
+
+;; Endpoint for updating a single invoice.
+;; PUT /api/invoices/<id>
+;; Calling this handler 'update' would shadow
+;; clojure.core/update
+(defn modify
+  [{:keys [route-params body]} connection]
+  ;; Validate input JSON
+  ;; should have only {paid: true}
+  ;; as user can only update an unpaid invoice to paid. 
+  (util/validate-spec body ::invoices-spec/invoice-paid)
+  (let [invoice-id (Integer/parseInt (:id route-params))]
+    (if-let [updated-invoice (invoices-db/update!
+                                connection
+                                invoice-id
+                                body)]
```

Comment:
It's generally a good idea to do a `select-keys` rather than passing on the entire body like this. Note that the `spec` will only verify that the specified keys are present, it will NOT fail if there are other keys present as well.